server/app.py

```diff
@@ -1,7 +1,183 @@
+
+
+
+def convert_to_point_geojson(data):
+    points = []
+    for b in data:
+        data = {
+            "type": "Feature",
+            "geometry": {
+                "type": "Point",
+                "coordinates": [b["longitude"], b["latitude"]]
+            }
+        }
+        points.append(data)
+    return {"type": "FeatureCollection", "features": points}
+
+
+def convert_to_polygon_geojson():
+    pass
+
+
+async def get_all_bins():
+    """
+        id": "lng:42,lat:13",
+        "anchorLongitude": 123,
+        "anchoLatitude": 12,
+        "totalBitrate": 1234,  
+        "totalAudioBitrate": 1324,
+        "count": 1234,
+        "points": [{longitude: 2, latitude: 3, bitrate: 2, audioBitrate: 14}, {...}, {...}]
+    """
+    bin_ref = db.collection("bins")
+    bins = bin_ref.get()
+
+    # get bins in json format
+    bins = [b.to_dict() for b in bins]
+
+    # preprocess all bins
+    preprocess = []
+    for b in bins:
+        _id = b['id']
+        anchor_longitude = b['anchorLongitude']
+        anchor_latitude = b['anchorLatitude']
+        total_bitrate = b['totalBitrate']
+        total_audiob_itrate = b['totalAudioBitrate']
+        count = b['count']
+        points = b['points']
+
+        avg_bitrate = 0
+        avg_audio_bitrate = 0
+        # using the center of the square as the point for showing up on the heatmap
+        longitude = anchor_longitude + SQUARE_SIZE
+        latitude = anchor_latitude - SQUARE_SIZE
+
+        if count > 0:
+            avg_bitrate = total_bitrate / count        
+            avg_audio_bitrate = total_audiob_itrate / count
+
+        preprocess.append({
+            longitude: longitude,
+            latitude: latitude,
+            avg_bitrate: avg_bitrate,
+            avg_audio_bitrate: avg_audio_bitrate
+        })
+    # use convext hull to process each bin if we like to use polygon for the heatmap
+    # ######################################
+
+    # convert the process data into an proper geojson  
+    formatted = convert_to_point_geojson(preprocess)
+    return formatted
+
+
+@app.route('/point_bitrate_heatmap', methods=['GET'])
+async def read():
+    """
+        read() : Fetches bin documents from Firestore collection as JSON.
+    """
+    try:
+        # get all bins
+        data = await get_all_bins()
+        return jsonify(data), 200
+    except Exception as e:
+        return f"An Error Occured: {e}"
+
+
+async def write_anchors(anchors):
+    for anchor in anchors:
+        anchor_longitude, anchor_latitude = anchor
+        data = {
+            "id": f"lng:{anchor_longitude},lat:{anchor_latitude}",
+            "anchorLongitude": anchor_longitude,
+            "anchoLatitude": anchor_latitude,
+            "totalBitrate": 0,  
+            "totalAudioBitrate": 0,
+            "count": 0,
+            "points": []
+        }
+        db.collection("bins").document(anchor).set(data)
+    return
+
+
+@app.route('/setbins', methods=['GET'])
```

Comment:
**Ignore if this is a dummy**, would this make more sense using POST if it is going to serve creation?